tests/unit/test_utilities.py

```diff
@@ -76,14 +92,110 @@ def test_open_gdal(self):
             access to a GeoTIFF file.
 
         """
-        uavsar_granule = ('tests/data/granules/uavsar/'
-                          'gulfco_32010_09045_001_090617_L090_CX_01_pauli.tif')
-
-        with OpenGDAL(uavsar_granule) as uavsar_gdal_object:
+        with OpenGDAL(self.uavsar_granule) as uavsar_gdal_object:
             gdal_metadata = uavsar_gdal_object.GetMetadata()
 
         self.assertDictEqual(gdal_metadata,
                              {'AREA_OR_POINT': 'Area',
                               'TIFFTAG_RESOLUTIONUNIT': '1 (unitless)',
                               'TIFFTAG_XRESOLUTION': '1',
                               'TIFFTAG_YRESOLUTION': '1'})
+
+    def test_is_geotiff(self):
+        """ Ensure that a file is correctly recognised as a GeoTIFF. """
+        with self.subTest('A GeoTIFF granule returns True'):
+            self.assertTrue(is_geotiff(self.uavsar_granule))
+
+        with self.subTest('A NetCDF-4 granules returns False'):
+            self.assertFalse(is_geotiff(self.sentinel_granule))
+
+    def test_get_files_from_unzipfiles(self):
```

Comment:
As stated in the PR description, I wrote these tests prior to the majority of the edits in the function itself. Hopefully, that ensures consistent behaviour with before.

Reply:
Although - I did tweak the tests after updating the function to cope with `.nc4` or `.tiff` extensions.